### PR TITLE
Added support for path “/:name/lastest”.

### PR DESCRIPTION
**Pull request by whitingj**
In npm 1.3.21 (and probably earlier versions than this) now request “/:name/lastest” when a “npm update” or “npm update <package>” commands are run.  Bumped reggie version.

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -187,9 +186,33 @@ server.get('/:name', function (req, res) {
...
+  var packageName = req.params.name;
+
+  var data = getPackageData(packageName);
+  if (!data) return notFound(res);
+
+  var version = data["dist-tags"]["latest"];
+  var versionMeta = data.versions[version];
+  if (!versionMeta) return notFound(res);
+
+  res.json(200, versionMeta);
```
Comment (bajtos) [inline server.js:213]: `/:name/:version` sends `versionMeta.data` in the response (see the code below), is it intentional that this method sends `versionMeta` instead?

**Comment by bajtos** [comment]
Hello @whitingj, thank you for the pull request. Please add a unit-test for this change to `test/npm-protocol.js`.

File: server.js (hunk 1 above)
Comment (whitingj) [inline server.js:213]: It is intentional.  server.get('/:name/:version') uses the data from data.packageMeta(name) and that requires the versionMeta.data.  

server.get('/:name/latest') piggies back off of server.get('/:name') which has already grabbed the information from versionMeta.data see below:

```
 versions.forEach(function(v) {
    versionsData[v] = meta.versions[v].data;
    times[v] = meta.versions[v].time;
  });
```

**Comment by whitingj** [comment]
I've added a unit test that install a package and then updates it and makes sure that npm doesn't fail.

File: test/npm-protocol.js [hunk 2]
```diff
@@ -86,6 +86,49 @@ describe('reggie npm server', function() {
...
+    helpers.cleanNpmCache();
+
+    helpers.runNpmInDirectory(
+      pkg.folder,
+      ['publish'],
+      function(err, stdout, stderr) {
+        if (err) return done(err);
+        // Give Reggie server some time to process the published package,
+        // because the response is sent before the processing is done
+        setTimeout(function() {
```
Comment (bajtos) [inline test/npm-protocol.js:106]: Please extract a helper method for L99-L106, possibly including L93.

Examples:

``` js:
helpers.givenAPublishedPackage(function(err, pkg) {
  // test code goes here
});

// alternative
var pkg = helpers.givenAPackage();
helpers.publish(pkg, function(err) {
    // test code goes here
});
```

Use the new test helper in the test above (`searches for a package that is already published`) too.

File: test/npm-protocol.js [hunk 3]
```diff
@@ -86,6 +86,49 @@ describe('reggie npm server', function() {
...
+              helpers.runNpmInDirectory(
+                installDir,
+                ['update', pkg.name],
+                function(err, stdout, stderr) {
+                  if (err) return done(err);
+                  //since there was no error npm exited with a valid return code and didn't fail
+                  done();
+                }
+              );
+
```
Comment (bajtos) [inline test/npm-protocol.js:125]: Extra empty line, please remove.

**Comment by bajtos** [comment]
> I've added a unit test that install a package and then updates it and makes sure that npm doesn't fail.

Nice :+1: 

However: have you seen the test to fail? I have removed the `/:name/latest` route added by your commit and the test was still passing.

File: server.js (hunk 1 above)
Comment (bajtos) [inline server.js:213]: I see. Could you rename the variable `versionMeta` to `versionData`, so that it's more clear what is the type of its value?

**Comment by alexgorbatchev** [comment]
npm 1.3 makes reggie barely useful, without being able to `npm update`. Is it possible to get this in ASAP?

**Comment by alexgorbatchev** [comment]
@whitingj, have you seen that last comment from @bajtos ?

**Pull request closed without merging**